pennylane/transforms/zx/reducer.py

```diff
@@ -0,0 +1,123 @@
+from pennylane.typing import PostprocessingFn
+
+has_pyzx = True
+try:
+    import pyzx
+except ImportError:
+    has_pyzx = False
+
+
+@transform
+def reduce_zx_calculus(tape: QuantumScript) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    """Reduce the circuit applying simplification rules based on the ZX calculus.
+
+    This transform returns an equivalent reduced version of the given quantum circuit, performing
+    a graph-theoretic simplification based on ZX calculus rules. It works as follows:
+
+        - convert the quantum circuit into the corresponding ``pyzx`` graph;
+
+        - apply ZX calculus simplification rules on the graph;
+
+        - convert the simplified ``pyzx`` graph back to its quantum circuit representation.
+
+    Args:
+        tape (QNode or QuantumScript or Callable): the input circuit to be transformed.
+
+    Returns:
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumScript], function]:
+        the transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
+    Raises:
+        ModuleNotFoundError: if the required ``pyzx`` Python package is not installed.
+
+    **Example:**
+
+    Consider the following QNode function as an example:
+
+    .. code-block:: python3
+
+        import pennylane as qml
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit(x, y):
+            qml.T(wires=0)
+            qml.Hadamard(wires=0)
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0,1])
+            qml.T(wires=0)
+            qml.RX(x, wires=1)
+            qml.RX(y, wires=1)
+            return qml.state()
+
+    To simplify the circuit using the ZX calculus rules you can do:
+
+    >>> new_circuit = qml.transforms.reduce_zx_calculus(circuit)
+    >>> print(qml.draw(new_circuit)(3.2, -2.2))
+    0: ────╭Z──S───────────┤  State
+    1: ──H─╰●──RZ(1.00)──H─┤  State
+
+    You can check that the final state is matching the one returned by the original circuit by:
+
+    >>> qml.math.allclose(circuit(3.2, -2.2), new_circuit(3.2, -2.2))
+    True
+
+    .. note::
+
+        This transform is designed to minimize T-count, and is not as effective at reducing the
+        number of two-qubit gates, such as CNOTs. However, its performance varies significantly
+        depending on the type of circuit. For example, you might see a substantial increase in CNOT
+        gates when optimizing a circuit composed primarily of Toffoli gates. Conversely, it tends
+        to perform much better on Trotterized chemistry circuits.
+
+    """
+
+    if not has_pyzx:  # pragma: no cover
+        raise ModuleNotFoundError(
+            "The `pyzx` package is required. You can install it by `pip install pyzx`."
+        )
+
+    zx_graph = qml.transforms.to_zx(tape)
+    pyzx.full_reduce(zx_graph)
+    zx_graph = pyzx.extract_circuit(zx_graph).to_graph()
+    qscript = qml.transforms.from_zx(zx_graph)
```

Comment:
Since they're the same module, can we promote `to_zx` and `from_zx` to be local relative imports?

```
from .converter import to_zx, from_zx
```